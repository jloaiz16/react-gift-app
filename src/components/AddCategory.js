import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue !== '') {
      setCategories((cat) => [inputValue, ...cat]);
      setInputValue('');
    } else {
      alert('Input empty, please add information');
    }
  };

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type='text' id='add' value={inputValue} onChange={handleChange} />
      <button>Add</button>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
