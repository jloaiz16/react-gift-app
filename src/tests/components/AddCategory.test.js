import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../../src/components/AddCategory';
import '@testing-library/jest-dom';

describe('<AddCategory /> suit test', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('Should render <AddCategory /> properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should change the input text value', () => {
    const input = wrapper.find('input');
    const value = 'Testing';
    input.simulate('change', { target: { value } });
    const inputAfter = wrapper.find('input');
    expect(inputAfter.prop('value')).toBe(value);
  });

  test('Shouldnt submit on form', () => {
    const jsdomAlert = window.alert;
    window.alert = () => {};
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    window.alert = jsdomAlert;
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('Should call setCateogies and clean input value', () => {
    const value = 'Testing';
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    expect(setCategories).toHaveBeenCalled();
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
