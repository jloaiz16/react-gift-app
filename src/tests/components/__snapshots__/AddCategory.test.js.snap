// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> suit test Should render <AddCategory /> properly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id="add"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button>
    Add
  </button>
</form>
`;
